operator/v1alpha1/types_kueue.go

```diff
@@ -0,0 +1,277 @@
+type KueueGangSchedulingPolicyOptions string
+
+const (
+	KueueGangSchedulingPolicyEvictNotReadyWorkloads KueueGangSchedulingPolicyOptions = "ByWorkload"
+	KueueGangSchedulingPolicyDisabled               KueueGangSchedulingPolicyOptions = "Disabled"
+)
+
+// +kubebuilder:validation:Enum=Parallel;Sequential
+type KueueGangSchedulingAdmissionOptions string
+
+const (
+	KueueGangSchedulingAdmissionOptionsSequential KueueGangSchedulingAdmissionOptions = "Sequential"
+	KueueGangSchedulingAdmissionOptionsParallel   KueueGangSchedulingAdmissionOptions = "Parallel"
+)
+
+// Kueue provides the ability to admit workloads all in one (gang admission)
+// and evicts workloads if they are not ready within a specific time.
+type KueueGangSchedulingPolicy struct {
+	// policy allows for changing the kinds of gang scheduling Kueue does.
+	// This is an optional field.
+	// The allowed values are ByWorkload and Disabled.
+	// The default value will be Disabled.
+	// ByWorkload allows for configuration how admission is performed
+	// for Kueue.
+	// +optional
+	Policy KueueGangSchedulingPolicyOptions `json:"policy"`
+	// byWorkload controls how admission is done.
+	// When admission is set to Sequential, only pods from the currently processing workload will be admitted.
+	// Once all pods from the current workload are admitted, and ready, Kueue will process the next workload.
+	// Sequential processing may slow down admission when the cluster has sufficient capacity for multiple workloads,
+	// but provides a higher guarantee of workloads scheduling all pods together successfully.
+	// When set to Parallel, pods from any workload will be admitted at any time.
+	// This may lead to a deadlock where workloads are in contention for cluster capacity and
+	// pods from another workload having successfully scheduled prevent pods from the current workload scheduling.
+	// +kubebuilder:validation:XValidation:rule="self.policy==ByWorkload",message="byWorkload is only valid if policy equals ByWorkload"
+	// +optional
+	ByWorkload KueueGangSchedulingAdmissionOptions `json:"byWorkload"`
+}
+
+// +kubebuilder:validation:Enum=QueueNameRequired;QueueNameOptional
+type QueueLabelNamePolicy string
+
+const (
+	QueueLabelNamePolicyRequired QueueLabelNamePolicy = "QueueNameRequired"
+	QueueLabelNamePolicyOptional QueueLabelNamePolicy = "QueueNameOptional"
+)
+
+type QueueLabelPolicy struct {
+	// queueLabelPolicy controls whether or not Kueue reconciles
+	// jobs that don't set the label kueue.x-k8s.io/queue-name.
+	// The allowed values are QueueNameRequired and QueueNameOptional.
+	// If set to QueueNameRequired, then those jobs will be suspended and never started unless
+	// they are assigned a queue and eventually admitted. This also applies to
+	// jobs created before starting the kueue controller.
+	// Defaults to QueueNameRequired; therefore, those jobs are not managed and if they are created
+	// unsuspended, they will start immediately.
```

Comment:
That sounds like what I'd expect of optional?

Reply:
No. QueueNameOptional would mean that Kueue will mutate all workloads in the namespace and add a kueue name label.

QueueNameRequired would mean that kueue would only manage a workload that has a certain label. Workloads that do not have this label will be skipped by Kueue.

Reply:
We may want to revisit the naming here as this is a little confusing to me.

I get the name required means that it will only process jobs that have the queue name labelled, but maybe optional is not really saying "manage everything whether it has a label or not"

Do you see what I mean?

Reply:
This is no longer relevant. Resolving.